### DISCUSSION TITLE
revalidatePath requires hard re-load for page navigation to work

### DISCUSSION THREAD

**Discussion by vikasrathod4299** [vercel/next.js#60262] [Help]
In my Next.js 14 application, I'm encountering an issue with `revalidatPath` listing page upon adding a record from the create page. after form submission, I've attempted to revalidate the data by putting `revalidatePath`. However, it fails to work properly, leading to infinite  loading during page navigation by clicking any `<Link>` or using `router.push`. Oddly enough, after a hard reload following the form submission,  the revalidation starts functioning as intended.

**Comment by zanzlender** [comment]
UPDATE: answer https://github.com/vercel/next.js/discussions/65487 

Did you maybe find a fix? It just happened to me 😂

I am fetching the Projects data in the server component on the Dashboard page. On the Create project page I use tRPC inside a client component to mutate the data. After it's done I router.push() to dashboard but it's not refetching... Same thing if you create a Project and through the navigation get to the Dashboard page
